### ISSUE TITLE
Windows and Mac client cannot connect to Internet with Wireguard

### ISSUE THREAD

**Issue by tiiiko** [trailofbits/algo#1691]
**OS / Environment (where do you run Algo on)**

> Linux (ubuntu 18.04) on server

**Cloud Provider (where do you deploy Algo to)**

> Azure

**Summary of the problem**

> When connecting through Wireguard with either Mac or PC, the client cannot connect to the Internet.

**Steps to reproduce the behavior**

> 

[It seems at least in earlier versions of AlgoVPN, there were special settings during ./Algo for Windows clients.](https://github.com/trailofbits/algo/issues/760) However, I installed AlgoVPN on my Ubuntu Server and saw no special settings related to Windows 10 or Wireguard when setting up ./Algo. 

After completion of ./Algo, I tried connecting my Mac with Wireguard and could not connect to the Internet when doing so. On later attempts, all my Apple devices (iPhone, iPad, Mac) could connect via the dedicated mobile configuration files. 

With my Windows 10 Laptop, it seems I need to connect via Wireguard and this connection is not working - i.e. after activating the connection, when hovering over the connection details it is displayed that the VPN user ("userWindows") cannot connect to the Wifi network ("TP-Link_XYZ"):

> TP-Link_XYZ
> Internet access
> 
> userWindows
> No Internet access

After a few minutes, the Internet connection (TP-Link_XYZ) changes from "connected, secured" to "No Internet, secured".

The same happens if I connect via my phone's network instead.

I setup Wireguard as described:

> Install the WireGuard VPN Client. Import the generated wireguard/<username>.conf file to your device, then setup a new connection with it.

Also, I tried deactivating the Windows Firewall and added an exception to the Virus scanner for Wireguard - the issue persisted after doing that.

Is this a known Issue? Is setting up the VPN connection manually instead of using Wireguard a good workaround? I ran into a "Policy match error" when attempting to do that on my Windows 10 laptop.

**Comment by jackivanov** [comment]
I'm not able to reproduce. Internet works if I deploy with Algo in Azure. Can you  provide more debug info, please? (eg: ping internal wireguard address, ping internal dns resolver's address, traceroute 8.8.8.8, etc..)

**Comment by davidemyers** [comment]
If this is a local install like #1690 maybe the Azure firewall needs to be changed (I don't remember if Azure has a mandatory external firewall)?

**Comment by tiiiko** [comment]
> I'm not able to reproduce. Internet works if I deploy with Algo in Azure. Can you provide more debug info, please? (eg: ping internal wireguard address, ping internal dns resolver's address, traceroute 8.8.8.8, etc..)

Thank you for your reply, I pinged all the addresses shown in WireGuard under Windows 10:

- Interface address: ping successful
- Interface DNS server (i.e. "local DNS resolver"): "request timed out"
- Peer endpoint (i.e. "public IP"): "request timed out"

I also executed a traceroute command:
- "request timed out" (30 times)


> If this is a local install like #1690 maybe the Azure firewall needs to be changed (I don't remember if Azure has a mandatory external firewall)?

Correct, this is a local install and it is the exact same installation you quoted.

Ok, firewall is a good keyword. I googled for "algoVPN firewall" and I found this link:
https://github.com/trailofbits/algo/blob/master/docs/firewalls.md

On this page (which is part of this github site) it says the following ports must be opened:

- 500
- 4500
- 51820

The last port is only used by WireGuard and was not covered by the install manuals I used. Now I activated it and I can use WireGuard on my Windows 10 laptops. 

Thank you very much!
